src/fs/driver/devfs/devfs_oldfs.c

```diff
@@ -133,7 +133,7 @@ static struct inode *devfs_lookup(char const *name, struct inode const *dir) {
 
 	cdevtab = get_cdev_tab();
 	bdevs = get_bdev_tab();
-	for (i = 0; i < MAX_BDEV_QUANTITY; i++) {
+	for (i = 0; i < MAX_BDEV_QUANTITY && bdevs[i]; i++) {
```

Comment:
But you will break here at the first `bdev[i] == NULL`, and other block devs will not be processed.

Reply:
It's better to skip null devices while processing all array up to the end

Reply:
I thought that reaching Null when iterating on the array means that there is no more block devs.
so I added the condition to make it break when reaching Null pointer.

Reply:
Some of these devices are possibly deleted, so we can skip them, so yes your new checking is pretty fine :)